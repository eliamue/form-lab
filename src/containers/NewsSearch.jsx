import React, { Component } from 'react';
import Search from '../components/news/Search';
import ArticleList from '../components/Article/NewsList';
import { fetchNewsArticleByTitle, fetchArticles } from '../services/newsApi';

export default class NewsSearch extends Component {
  state = {
    loading: true,
    articles: [],
    articleTitle: '',
  };

  async componentDidMount() {
    const articles = await fetchArticles();
    this.setState({ articles, loading: false });
  }

  handleArticleTitleChange = (event) => {
    this.setState({ articleTitle: event.target.value });
  };

  handleSubmit = async (event) => {
    event.preventDefault();

    this.setState({ loading: true });
    const articles = await fetchNewsArticleByTitle(this.state.articleTitle);
    this.setState({ articles, loading: false });
  };

  render() {
    const { loading, articles, articleTitle } = this.state;

    if(loading) return <p>Loading</p>;
    return (
      <div>
        <Search
          articleTitle={articleTitle}
          onArticleTitleChange={this.handleArticleTitleChange}
          onSubmit={this.handleSubmit}
        />
        <ArticleList articles={articles} />
      </div>
    );
  }
}
