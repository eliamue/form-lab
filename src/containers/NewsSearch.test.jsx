import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import NewsSearch from './NewsSearch';

describe('NewsSearch', () => 
  it('displays a list of news articles', async () => {
    render(<NewsSearch />);

    screen.getByText('Loading');
    const ul = await screen.findByRole('list', { name: 'articles' });
    expect(ul).not.toBeEmptyDOMElement();

  })
);
