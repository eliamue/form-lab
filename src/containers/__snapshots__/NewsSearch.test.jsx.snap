// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsSearch displays a list of news articles 1`] = `
Array [
  <figcaption>
    Sharknado: The Complete 6-Movie Saga in HD Digital for $4.99 at Microsoft
     - 
    cheebong
  </figcaption>,
  <p>
    Microsoft has Sharknado: The Complete Saga for $4.99. TODAY only. It is NOT Movies Anywhere, so no port. 
 
Own all 6 Sharknado films â€“ the entire saga! Take the journey from the first storm, crushi ...
  </p>,
]
`;
